src/test/java/htsjdk/tribble/annotation/StrandTest.java

```diff
@@ -0,0 +1,94 @@
+import org.testng.annotations.Test;
+
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Unit tests for {@link Strand}.
+ * <p>
+ *     Makes sure that {@link Strand} will behave according to contract in run-time.
+ * </p>
+ */
+public class StrandTest {
+
+    @Test
+    public void testEncodeReturnNoNulls() {
+        for (final Strand subject : Strand.values()) {
+            Assert.assertNotNull(subject.encode());
+        }
+    }
+
+    @Test
+    public void testEncodeDecode() {
+        for (final Strand subject : Strand.values()) {
+            final String encode = subject.encode();
+            final Strand back = Strand.decode(encode);
+            Assert.assertSame(back, subject);
+        }
+    }
+
+    @Test
+    public void testEncodeDecodeAsChar() {
+        for (final Strand subject : Strand.values()) {
+            final char encode = subject.encodeAsChar();
+            final Strand back = Strand.decode(encode);
+            Assert.assertSame(back, subject);
+        }
+    }
+
+    @Test
+    public void testEncodeUniqueness() {
+        Assert.assertEquals(Stream.of(Strand.values())
+                .map(Strand::encode)
+                .collect(Collectors.toSet()).size(), Strand.values().length);
+    }
+
+    @Test
+    public void testEncodeAsCharUniqueness() {
+        Assert.assertEquals(Stream.of(Strand.values())
+                .map(Strand::encodeAsChar)
+                .collect(Collectors.toSet()).size(), Strand.values().length);
+    }
+
+    @Test
+    public void testEncodeAsStringAndCharAreEquivalent() {
+        for (final Strand subject : Strand.values()) {
+            final String str = subject.encode();
+            final char ch = subject.encodeAsChar();
+            Assert.assertEquals(str, String.valueOf(ch));
+        }
+    }
+
+    @Test(dataProvider = "invalidEncodingStringsData", expectedExceptions = IllegalArgumentException.class)
+    public void testDecodeOfInvalidEncodingStrings(final String encoding) {
+        Strand.decode(encoding);
+    }
+
+    @Test(dataProvider = "invalidEncodingCharsData", expectedExceptions = IllegalArgumentException.class)
+    public void testDecodeOfInvalidEncodingChars(final char encoding) {
+        Strand.decode(encoding);
+    }
+
+    @DataProvider(name = "invalidEncodingStringsData")
+    public Object[][] invalidEncodingStringsData() {
+        return Stream.of("", null, "+++", "---", "POSITIVE",
+                "NEGATIVE", "NONE",
+                "FORWARD", "REVERSE",
+                " +", "- ", "p", "m", "x")
+                .map(s -> new Object[] { s}).toArray(Object[][]::new);
+    }
+
+    @DataProvider(name = "invalidEncodingCharsData")
+    public Object[][] invalidEncodingCharsData() {
+        final char[] testCases = new char[] {'c', '\t', '\n', (char) -1, 'p', 'n', 'x' };
+        return IntStream.range(0, testCases.length)
+                .mapToObj(i -> testCases[i])
```

Comment:
Do you need `IntStream` here? It looks like you could use `Stream.of()` as do you for the string data.

Reply:
Yeah I guess i should just Use a ```Character[]``` array rather than a ```char``` array, perhaps implicitly boxed by Stream.of(Object ... o)